Sources/Action.swift

```diff
@@ -174,6 +184,36 @@ extension Action: ActionProtocol {
 	}
 }
 
+extension ActionProtocol where Input == Void {
+	/// Initializes an action that uses an optional property for its input,
+	/// and is disabled whenever the input is nil. When executed, a SignalProducer
+	/// is created with the current value of the input.
+	///
+	/// - parameters:
+	///   - input: An Optional property whose current value is used as input
+	///            whenever the action is executed. The action is disabled
+	///            whenever the value is nil.
+	///   - execute: A closure to return a new SignalProducer based on the
+	///              current value of `input`.
+	public init<P: PropertyProtocol, T>(input: P, _ execute: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T? {
+		self.init(enabledIf: input.map { $0 != nil }) {
+			execute(input.value!)
```

Comment:
Thinking about this more, I don't think this is safe.

The property's value could change between the enabled check and the time execute is called. If the value has switched back to `nil`, then this will assert.

I think this will require some changes to `apply` and maybe a different initializer. I don't see how we can make the current approach work.